samples/Ralltiir/Ralltiir.Server/Controllers/AuthorizationController.cs

```diff
@@ -0,0 +1,397 @@
+                case ConsentTypes.Explicit when authorizations.Any() && !request.HasPrompt(Prompts.Consent):
+                    var principal = await _signInManager.CreateUserPrincipalAsync(user);
+
+                    // Note: in this sample, the granted scopes match the requested scope
+                    // but you may want to allow the user to uncheck specific scopes.
+                    // For that, simply restrict the list of scopes before calling SetScopes.
+                    principal.SetScopes(request.GetScopes());
+                    principal.SetResources(await _scopeManager.ListResourcesAsync(principal.GetScopes()).ToListAsync());
+
+                    // Automatically create a permanent authorization to avoid requiring explicit consent
+                    // for future authorization or token requests containing the same scopes.
+                    var authorization = authorizations.LastOrDefault();
+                    if (authorization == null)
+                    {
+                        authorization = await _authorizationManager.CreateAsync(
+                            principal: principal,
+                            subject  : await _userManager.GetUserIdAsync(user),
+                            client   : await _applicationManager.GetIdAsync(application),
+                            type     : AuthorizationTypes.Permanent,
+                            scopes   : principal.GetScopes());
+                    }
+
+                    principal.SetAuthorizationId(await _authorizationManager.GetIdAsync(authorization));
+
+                    foreach (var claim in principal.Claims)
+                    {
+                        claim.SetDestinations(GetDestinations(claim, principal));
+                    }
+
+                    return SignIn(principal, OpenIddictServerAspNetCoreDefaults.AuthenticationScheme);
+
+                // At this point, no authorization was found in the database and an error must be returned
+                // if the client application specified prompt=none in the authorization request.
+                case ConsentTypes.Explicit   when request.HasPrompt(Prompts.None):
+                case ConsentTypes.Systematic when request.HasPrompt(Prompts.None):
+                    return Forbid(
+                        authenticationSchemes: OpenIddictServerAspNetCoreDefaults.AuthenticationScheme,
+                        properties: new AuthenticationProperties(new Dictionary<string, string>
+                        {
+                            [OpenIddictServerAspNetCoreConstants.Properties.Error] = Errors.ConsentRequired,
+                            [OpenIddictServerAspNetCoreConstants.Properties.ErrorDescription] =
+                                "Interactive user consent is required."
+                        }));
+
+                // In every other case, render the consent form.
+                default: return Json(new AuthorizeViewModel
```

Comment:
The authorization endpoint is not expected to return a JSON payload, it's an interactive endpoint.

We have two options here:
  - Either you remove all the consent view stuff and just relies on the `prompt=none` support (see the implicit flow sample for a simplified version). You can use the `Forbid` dance to trigger a redirection to the client to indicate consents are not supported.
  - Or you return the "SPA" loading page and handle that client-side.

Reply:
Or 3rd option: you return a special view that loads your SPA and flows the OIDC parameters in a way that can be interpreted by the client-side (e.g data attributes, in a HTML block).